R/dummy_data.R

```diff
@@ -202,6 +202,255 @@ dummy_diag <- function(nid = 5, nrow = 50, ipdiagnosis = TRUE, diagnosis_type =
+#'    - 65: Did not Return from Pass/Leave
+#'    - 66: Died While on Pass/Leave
+#'    - 67: Suicide out of Facility (does not exist in GEMINI data)
+#'    - 72: Died in Facility
+#'    - 73: Medical Assistance in Dying (MAID)
+#'    - 74: Suicide in Facility
+#' - `alc_service_transfer_flag` (`character`): Variable indicating whether
+#' patient was transferred to an alternate level of care (ALC) during their
+#' hospital stay. Coding is messy and varies across sites. Possible values are:
+#'    - Missing: `NA`, `""`
+#'    - True: `"TRUE"/"true"/"T"`, `"y"/"Y"`, `"1"/"99"`, `"ALC"`
+#'    - False: `"FALSE"/"false"`, `"N"`, `"0"`, `"non-ALC"`
+#' Some entries with missing `alc_service_transfer_flag` can be inferred based
+#' on value of `number_of_alc_days` (see below)
+#' - `number_of_alc_days` (`integer`): Number of days spent in ALC (rounded to
+#' nearest integer). If `number_of_alc_days = 0`, no ALC occurred;
+#' if `number_of_alc_days > 0`, ALC occurred.
+#' Note that days spent in ALC should usually be < length of
+#' stay. However, due to the fact that ALC days are rounded up, it's possible
+#' for `number_of_alc_days` to be larger than `los_days_derived`.
+#'
+#' @importFrom sn rsn
+#' @importFrom MCMCpack rdirichlet
+#' @importFrom lubridate ymd_hm
+#' @export
+#'
+#' @examples
+#' # Simulate 10,000 encounters from 10 hospitals for fiscal years 2018-2020.
+#' ipadmdad <- dummy_ipadmdad(n = 10000, n_hospitals = 10, time_period = c(2018, 2020))
+#'
+#'
+dummy_ipadmdad <- function(n = 1000,
+                           n_hospitals = 10,
+                           time_period = c(2015, 2023)) {
+
+
+  ############### CHECKS: Make sure n is at least n_hospitals * length(time_period)
+  if (n < n_hospitals*length(time_period)){
+    stop("Invalid user input.
+    Number of encounters `n` should at least be equal to `n_hospitals` * `length(time_period)`")
+  }
+
+  ############### PREPARE OUTPUT TABLE ###############
+  ## create all combinations of hospitals and fiscal years
+  hospital_num <- seq(1,n_hospitals,1)
+  year <- seq(time_period[1],time_period[2],1)
+
+  data <- expand.grid(hospital_num = hospital_num, year = year) %>% data.table()
+
+  # randomly draw number of encounters per hospital*year combo
+  # make sure they add up to desired total number of encounters
+  data[ , n := rmultinom(1, n, rep.int(1 / nrow(data), nrow(data)))]
+  sum(data$n)
+
+  # blow up row number according to encounter per combo
+  data <- data[rep(seq(nrow(data)), data$n),]
+
+  # turn year variable into actual date by randomly drawing date_time
+  add_random_datetime <- function(year) {
+    start_date <- paste0(year, "-04-01 00:00 UTC") # start each fisc year on Apr 1
+    end_date <- paste0(year+1, "-03-31 23:59 UTC")   # end of fisc year
+
+    random_datetime <- format(as.POSIXct(runif(length(year), as.POSIXct(start_date), as.POSIXct(end_date))),
+                              origin = "1970-01-01", format = "%Y-%m-%d %H:%M") #
```

Comment:
sorry @loffleraSMH the origin is an argument to `as.POSIXct`, so in other words the line should look like this:

```
format(as.POSIXct(runif(length(year), as.POSIXct(start_date), as.POSIXct(end_date)), origin = "1970-01-01"), format = "%Y-%m-%d %H:%M")
```

Reply:
Oops, sorry - good catch! Fixed it. Also ran `styler` so there are few other changes too but no other code-related updates.